### PR TITLE
Added troubleshooting entry for SDL2 build error

### PR DESCRIPTION
**Pull request by sarcXD**
This is a guide for a workaround for an issue I came across here #1433  and is hopefully detailed enough such that in case someone comes across this issue they can resolve it easily.  
@heinezen I am tagging you here, since you were the one who responded to my issue.

### CODE REVIEW

File: doc/troubleshooting.md [hunk 1]
```diff
@@ -26,3 +26,16 @@ A workaround would be to make a backup of your AGE2 directory and let the conver
 backup at subfolder `AGE2/resources` delete all files ***except*** folders. Another workaround would be to
 backup your AGE2 folder and redownload it to have a clean install. After conversion you can replace
 it with the backup.
+
+## Installation  
+
+### Cannot specify compile definitions for target "SDL2::SDL2" which is not built by this project  
+
+This specific error is specific to a few operating systems, and the main cause
```
Comment (simonsan) [inline doc/troubleshooting.md:34]:
```suggestion
This error is specific to a few operating systems, and the main cause
```

File: doc/troubleshooting.md [hunk 2]
```diff
@@ -26,3 +26,16 @@ A workaround would be to make a backup of your AGE2 directory and let the conver
...
+is mainly your SDL2 version likely not being the latest version. There is an
+indepth discussion about this
+[here](https://discourse.libsdl.org/t/how-is-sdl2-supposed-to-be-used-with-cmake/31275/16) 
+but the main solution is that you need to **compile SDL2 and SDL2-image from
+source**.  
+The source you should use should be from github, the latest version will allow
```
Comment (simonsan) [inline doc/troubleshooting.md:40]:
```suggestion
The source should be from github, the latest version will allow
```

File: doc/troubleshooting.md [hunk 3]
```diff
@@ -26,3 +26,16 @@ A workaround would be to make a backup of your AGE2 directory and let the conver
...
+your cmake to work with the SDL2::SDL2 call.
```
Comment (simonsan) [inline doc/troubleshooting.md:41]:
```suggestion
your CMake to work with the SDL2::SDL2 call.
```

**Review by heinezen — APPROVED** [review]
I've updated the text to mention the specific SDL version that start including the new CMake files.

We can include this troubleshooting section even if we implement the solution discussed in https://github.com/SFTtech/openage/issues/1433#issuecomment-975631219. At least until all repos have caught up with the new SDL releases.

**Pull request merged**